Add Ripple provider (#366)
Web3Auth/web3auth-docs: docs/connect-blockchain/ripple.mdx

> ---

Reviewer: The file name should be `xrpl.mdx`.

> title: Integrate Web3Auth with the Ripple Blockchain
> sidebar_label: Ripple
> image: "content-hub/guides/banners/ripple.png"

Reviewer: ```suggestion
title: Integrate Web3Auth with the XRPL Blockchain
sidebar_label: XRPL
image: "content-hub/guides/banners/xrpl.png"
```

displayed_sidebar: docs
keywords: [ripple, web3auth, authentication, blockchain]

> description: "Integrate Web3Auth with the Ripple Blockchain | Documentation - Web3Auth"

Reviewer: ```suggestion
description: "Integrate Web3Auth with the XRPL Blockchain | Documentation - Web3Auth"
```

---

import GetUserInfoSnippet from "@site/src/common/docs/_get-userinfo.mdx";

---

> The Web3Auth Web SDK offers a standard provider for accessing the user's private key on non-EVM chains like [Ripple](https://ripple.com/). To enable a

Reviewer: ```suggestion
The Web3Auth Web SDK offers a standard provider for accessing the user's private key on non-EVM chains like [XRPL](https://xrpl.org/). To enable a
```

sending transactions, and reading from and writing to smart contracts. Here are some methods to help you quickly get started.

## Installation

```bash npm2yarn
npm install --save xrpl @web3auth/xrpl-provider @web3auth/modal @web3auth/openlogin-adapter @web3auth/base
```

## Initializing Provider

---

```js
import { Web3Auth } from "@web3auth/modal";
import { CHAIN_NAMESPACESS } from "@web3auth/base";
import { OpenloginAdapter } from "@web3auth/openlogin-adapter";
import { XrplPrivateKeyProvider, getXRPLChainConfig } from "@web3auth/xrpl-provider";
import { convertStringToHex, Payment, xrpToDrops } from "xrpl";

const web3auth = new Web3Auth({
  chainConfig: {
    chainNamespace: CHAIN_NAMESPACES.OTHER,
  },
  clientId = "YOUR_WEB3AUTH_CLIENT_ID", // get from https://dashboard.web3auth.io
  web3AuthNetwork = "cyan", // testnet, mainnet, cyan, aqua or celeste
});

const xrplProvider = new XrplPrivateKeyProvider({
  config: {
    chainConfig: getXRPLChainConfig("testnet"), // devnet, testnet, mainnet
  },
});

const adapter = new OpenloginAdapter({
  adapterSettings: {

>     provider: xrplProvider, // <-- Injecting the Ripple provider

Reviewer: ```suggestion
    provider: xrplProvider, // <-- Injecting the XRPL provider
```

  },
});
web3AuthInstance.configureAdapter(adapter);

await web3auth.initModal();

> const rippleProvider = await web3auth.connect(); // web3auth.provider

Reviewer: ```suggestion
const xrplProvider = await web3auth.connect(); // web3auth.provider
```

```

## Get User Info

---

<GetUserInfoSnippet />

## Get Account and Balance

---

```js
try {
  // rippleProvider is from above
  const accounts = await rippleProvider.request<string[]>({
    method: "xrpl_getAccounts",
  });

  if (accounts) {

>     const accInfo = (await rippleProvider.request({

Reviewer: ```suggestion
    const accInfo = (await xrplProvider.request({
```

      method: "account_info",
      params: [
        {
          account: accounts[0],
          strict: true,
          ledger_index: "current",
          queue: true,
        },
      ],
    })) as Record<string, Record<string, string>>;
    console.log("xrpl account info", accInfo);
    // Ripple Account
    const account = accInfo?.account_data?.Account;
    // Balance
    const balance = accInfo?.account_data?.Balance;
  } else {
    console.log("No accounts found, please report this issue.");
  }
} catch (error) {
  console.error("Error", error);
}
```

## Sign Transaction

---

```js
try {
  const accounts = await provider.request<string[]>({
    method: "xrpl_getAccounts",
  });

  if (accounts && accounts.length > 0) {
    const tx: Payment = {
      TransactionType: "Payment",
      Account: accounts[0] as string,
      Amount: xrpToDrops(2),
      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX", // Destination address
    };
    const txSign = await provider.request({

>       method: "ripple_signTransaction",

Reviewer: ```suggestion
      method: "xrpl_signTransaction",
```

      params: {
        transaction: tx,
      },
    });
    console.log("txRes", txSign);
  } else {
    console.log("failed to fetch accounts");
  }
} catch (error) {
  console.log("error", error);
}
```

## Send Transaction

---

```js
try {
  const accounts = await provider.request<string[]>({
    method: "xrpl_getAccounts",
  });

  if (accounts && accounts.length > 0) {
    const tx: Payment = {
      TransactionType: "Payment",
      Account: accounts[0] as string,
      Amount: xrpToDrops(2),
      Destination: "rJAHHPYmy4g3h7kzfj2Mzm2nHwpKuVdEvX",
    };
    const txSign = await provider.request({

>       method: "ripple_submitTransaction",

Reviewer: ```suggestion
      method: "xrpl_submitTransaction",
```
